solidity/contracts/Coordinator.sol

```diff
@@ -167,18 +158,61 @@ contract Coordinator is ChainlinkRequestInterface, CoordinatorInterface {
     emit NewServiceAgreement(serviceAgreementID, _requestDigest);
   }
 
+  function fulfillOracleRequest(
+    bytes32 _requestId,
+    bytes32 _data
+  )
+    external
+    isValidResponse(_requestId)
+    returns (bool)
+  {
+    callbacks[_requestId].responses[msg.sender] = uint256(_data);
+
+    Callback memory callback = callbacks[_requestId];
+    callbacks[_requestId].responseCount += 1;
+    uint256 responseCount = callbacks[_requestId].responseCount;
+    bytes32 sAId = callbacks[_requestId].sAId;
+    address[] memory oracles = serviceAgreements[sAId].oracles;
+    if (oracles.length > responseCount) {
+      return true;
+    }
+
+    uint256 result;
+    uint256 oraclePayment = callback.amount.div(oracles.length);
+    for (uint i = 0; i < responseCount; i++) {
+      result = result.add(callbacks[_requestId].responses[oracles[i]]);
+      delete callbacks[_requestId].responses[oracles[i]];
+      withdrawableTokens[oracles[i]] = withdrawableTokens[oracles[i]].add(oraclePayment);
+    }
+    result = result.div(responseCount);
+    delete callbacks[_requestId];
+    return callback.addr.call(callback.functionId, _requestId, result); // solium-disable-line security/no-low-level-calls
+  }
+
+  function withdraw(address _recipient, uint256 _amount) 
+    external
+	hasAvailableFunds(_amount)
+  {
+    withdrawableTokens[msg.sender] = withdrawableTokens[msg.sender].sub(_amount);
+    assert(LINK.transfer(_recipient, _amount));
```

Comment:
I think this should be require rather than assert since assert uses up all gas given in a tx.